search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java

```diff
@@ -198,33 +205,157 @@ void shouldReturnBadGatewayResponseWhenNoResponseFromService() throws IOExceptio
         assertEquals(HTTP_BAD_GATEWAY, gatewayResponse.getStatusCode());
     }
 
+    @Test
+    void shouldReturnApplicationJsonIfNoAcceptHeaderProvided() throws IOException {
+        prepareRestHighLevelClientOkResponse();
+        handler.handleRequest(getInputStream(), outputStream, mock(Context.class));
+
+        GatewayResponse<String> gatewayResponse = GatewayResponse.fromOutputStream(outputStream, String.class);
+        assertThat(gatewayResponse.getHeaders().get("Content-Type"), is(equalTo("application/json; charset=utf-8")));
+    }
+
+    @Test
+    void shouldReturnApplicationJsonIfAcceptApplicationJsonProvided() throws IOException {
+        prepareRestHighLevelClientOkResponse();
+        handler.handleRequest(getRequestInputStreamAccepting("application/json"), outputStream, mock(Context.class));
+
+        GatewayResponse<String> gatewayResponse = GatewayResponse.fromOutputStream(outputStream, String.class);
+        assertThat(gatewayResponse.getHeaders().get("Content-Type"), is(equalTo("application/json; charset=utf-8")));
+    }
+
+    @Test
+    void shouldReturnTextCsvIfAcceptTextProvided() throws IOException {
+        prepareRestHighLevelClientOkResponse(List.of(csvWithFullDate(), csvWithYearOnly()));
+        handler.handleRequest(getRequestInputStreamAccepting("text/*"), outputStream, mock(Context.class));
+
+        GatewayResponse<String> gatewayResponse = GatewayResponse.fromOutputStream(outputStream, String.class);
+        assertThat(gatewayResponse.getHeaders().get("Content-Type"), is(equalTo("text/csv; charset=utf-8")));
+    }
+
+    @Test
+    void shouldReturnTextCsvIfAcceptTextCsvProvided() throws IOException {
+        prepareRestHighLevelClientOkResponse(List.of(csvWithFullDate(), csvWithYearOnly()));
+        handler.handleRequest(getRequestInputStreamAccepting("text/csv"), outputStream, mock(Context.class));
+
+        GatewayResponse<String> gatewayResponse = GatewayResponse.fromOutputStream(outputStream, String.class);
+        assertThat(gatewayResponse.getHeaders().get("Content-Type"), is(equalTo("text/csv; charset=utf-8")));
+    }
```

Comment:
ParameterisedTest? Also text/html…

Reply:
Added parameterised tests. What is requirement for `text/html`?

Reply:
Not sure…I think I might just expect json…